romancal/resample/resample.py

```diff
@@ -385,63 +398,65 @@ def resample_variance_array(self, name, output_model):
         inverse_variance_sum = np.full_like(output_model.data.value, np.nan)
 
         log.info(f"Resampling {name}")
-        for model in self.input_models:
-            variance = getattr(model, name)
-            if variance is None or variance.size == 0:
-                log.debug(
-                    f"No data for '{name}' for model "
-                    f"{repr(model.meta.filename)}. Skipping ..."
-                )
-                continue
-            elif variance.shape != model.data.shape:
-                log.warning(
-                    f"Data shape mismatch for '{name}' for model "
-                    f"{repr(model.meta.filename)}. Skipping..."
+        with self.input_models:
+            for i, model in enumerate(self.input_models):
+                variance = getattr(model, name)
+                if variance is None or variance.size == 0:
+                    log.debug(
+                        f"No data for '{name}' for model "
+                        f"{repr(model.meta.filename)}. Skipping ..."
+                    )
+                    continue
+                elif variance.shape != model.data.shape:
+                    log.warning(
+                        f"Data shape mismatch for '{name}' for model "
+                        f"{repr(model.meta.filename)}. Skipping..."
+                    )
+                    continue
```

Comment:
For what it's worth, these could equally well be failures & exceptions.

Reply:
I opened an issue to track this change: https://github.com/spacetelescope/romancal/issues/1308